### ISSUE TITLE
Firmware update on Logical enclosure fails

### ISSUE THREAD

**Issue by ramakrishnan20** [HewlettPackard/POSH-HPOneView#465]
### Expected Behavior
Upgrading the firmware of logical enclosure with the Custom SPP created from Base SPP should be successful

### Actual Behavior

Firmware upgrade of logical enclosure failed wherein the firmware bundle was custom SPP created from base SPP.
 We created a custom SPP in the oneview from the base SPP and tried to update the firmware of the logical enclosure with this newly created custom SPP using the oneview 5.00 powershell library.
But it failed and threw the following error that 
“The provided Baseline object customSPP_1 is not a valid resource object.  Only an SPP can be used to update firmware. Please correct your input value.”

While it works if we manually update the firmware (same customSPP) from the oneview->logicalEnclosure->Update Firmware.

### Steps to reproduce

```PowerShell
1. Refer the attached file for command trace output.

2. Code details here:
$customSPPObj= Get-HPOVBaseline -Name "customSPP_1" 
         
            try{
                Get-HPOVCommandTrace -ScriptBlock {$task = $lgEnclosure | Update-HPOVLogicalEnclosureFirmware -Baseline $customSPPObj -FirmwareUpdateProcess SharedInfrastructureOnly -InterconnectActivationMode Orchestrated}
                $taskList += $task
            }
            catch{
                Write-Host "Updating firmware on Logial Enclosure $enc failed. Error: $_"
            }


# 3. Remove this comment block before submitting.
```
[2003120513_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPOneView/files/4325355/2003120513_HPOV_transcript.log)
### Version Information

```powershell
 Get-HPOVVersion

composer.cs750.hpe.net              LibraryVersion Path
----------------------              -------------- ----
ApplianceVersion: 5.00.00.400525.00 5.0.2226.2609  C:\Program Files (x86)\WindowsPowerShell\Modules\HPOneView.500
```
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  

```powershell
Get-HPOVVersion -ApplianceVer

composer.cs750.hpe.net              LibraryVersion Path
----------------------              -------------- ----
ApplianceVersion: 5.00.00.400525.00 5.0.2226.2609  C:\Program Files (x86)\WindowsPowerShell\Modules\HPOneView.500
````
Output from `$PSVersionTable` on your Windows Host:
```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.14393.3471
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.3471
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by ChrisLynchHPE** [comment]
Hmm...  Can you please provide the following:

```PowerShell
$customSPPObj= Get-HPOVBaseline -Name "customSPP_1" 

$customSPPObj | Get-Member -view all
```

You should get the following output:

```powershell


   TypeName: HPOneView.Appliance.Baseline

Name                MemberType Definition
----                ---------- ----------
Equals              Method     bool Equals(System.Object obj)
GetHashCode         Method     int GetHashCode()
GetType             Method     type GetType()
ToString            Method     string ToString()
ApplianceConnection Property   HPOneView.Library.ApplianceConnection ApplianceConnection {get;set;}
BaselineShortName   Property   string BaselineShortName {get;set;}
BundleSize          Property   long BundleSize {get;set;}
BundleType          Property   HPOneView.Appliance.BundleType BundleType {get;set;}
Category            Property   string Category {get;}
Created             Property   datetime Created {get;set;}
Description         Property   string Description {get;set;}
ETag                Property   string ETag {get;set;}
FwComponents        Property   System.Collections.Generic.IList[HPOneView.Appliance.Baseline+FwComponent] FwComponen...
HotFixes            Property   System.Collections.Generic.IList[HPOneView.Appliance.Baseline+Hotfix] HotFixes {get;}
HPSumVersion        Property   string HPSumVersion {get;set;}
IsoFileName         Property   string IsoFileName {get;set;}
Locations           Property   System.Collections.Generic.IList[string] Locations {get;}
Mirrorlist          Property   System.Collections.Generic.IList[HPOneView.Appliance.Baseline+Mirror] Mirrorlist {get;}
Modified            Property   datetime Modified {get;set;}
Name                Property   string Name {get;}
ParentBundle        Property   HPOneView.Appliance.Baseline+ParentBaseline ParentBundle {get;set;}
ReleaseDate         Property   datetime ReleaseDate {get;set;}
ResourceId          Property   string ResourceId {get;set;}
ResourceState       Property   string ResourceState {get;set;}
ScopesUri           Property   uri ScopesUri {get;set;}
State               Property   string State {get;set;}
Status              Property   string Status {get;set;}
SupportedLanguages  Property   string SupportedLanguages {get;set;}
SupportedOSList     Property   System.Collections.Generic.IList[string] SupportedOSList {get;}
Uri                 Property   uri Uri {get;set;}
Uuid                Property   string Uuid {get;set;}
Version             Property   SemanticVersion Version {get;set;}
XmlKeyName          Property   string XmlKeyName {get;set;}
```

**Comment by ramakrishnan20** [comment]
Yes, we get that list.
Please the command output to view the object content.


![image](https://user-images.githubusercontent.com/62111482/76589794-3833f500-6511-11ea-9858-6fb09fb42326.png)

**Comment by ChrisLynchHPE** [comment]
The issue is with the baseline object validation logic.  This will be fixed in the next library update.

**Comment by ChrisLynchHPE** [comment]
This issue isn't quite completely fixed yet.  `Update-HPOVLogicalEnclosureFirmware` doesn't validate properly for `[HPOneView.Appliance.BundleType]::Custom`.  This will be fixed in the next update.

**Comment by ChrisLynchHPE** [comment]
This issue isn't quite completely fixed yet.  `Update-HPOVLogicalEnclosureFirmware` doesn't validate properly for `[HPOneView.Appliance.BundleType]::Custom`.  This will be fixed in the next update.

**Comment by ChrisLynchHPE** [comment]
This issue isn't quite completely fixed yet.  `Update-HPOVLogicalEnclosureFirmware` doesn't validate properly for `[HPOneView.Appliance.BundleType]::Custom`.  This will be fixed in the next update.

**Comment by ChrisLynchHPE** [comment]
This is addressed in [Release 5.0.2400.1845](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v5.0.2400.1845) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.500/5.0.2400.1845).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.  If you wish, feel free to re-open to discuss further.
